src/stringformat.c

```diff
@@ -27,6 +27,41 @@
 #include "stringformat-private.h"
 #include "general-private.h"
 
+static GpStringFormat *stringFormatDefault = NULL;
+static GpStringFormat *stringFormatTypographic = NULL;
+
+GpStatus
+gdip_create_generic_stringformats ()
+{
+	GpStatus status;
+	const int typographicFlags = StringFormatFlagsNoFitBlackBox | StringFormatFlagsLineLimit | StringFormatFlagsNoClip;
+
+	status = GdipCreateStringFormat (0, 0, &stringFormatDefault);
+	if (status != Ok)
+		return status;
+	
+	status = GdipCreateStringFormat (typographicFlags, 0, &stringFormatTypographic);
+	if (status != Ok)
+		return status;
+
+	stringFormatTypographic->trimming = StringTrimmingNone;
+	return Ok;
+}
```

Comment:
On a second thought, can we make `stringFormatGeneric` and `stringFormatTypographic` declared as `GpStringFormat` instead of `GpStringFormat *`? Then you would not need to allocate and free memory and you can avoid a lot of error checking.